### ISSUE TITLE
build fails on camlp4 tutorial

### ISSUE THREAD

**Issue by agarwal** [ocaml/v2.ocaml.org#617]
```
ocamlfind ocamlc -annot -I +camlp4 camlp4lib.cma -pp camlp4orf -c site/learn/tutorials/camlp4_3.10/pa_foreach3.ml -o site/learn/tutorials/camlp4_3.10/pa_foreach3.cmo
File "site/learn/tutorials/camlp4_3.10/pa_foreach3.ml", line 29, characters 15-31:
Error: This expression has type string but an expression was expected of type
         Camlp4.PreCast.Syntax.Ast.patt = Camlp4.PreCast.Ast.patt
make[3]: *** [site/learn/tutorials/camlp4_3.10/pa_foreach3.cmo] Error 2
make[2]: *** [gen_md] Error 2
make[1]: *** [staging] Error 2
make: *** [local] Error 2
```

**Comment by dsheets** [comment]
`opam list -i`?

**Comment by agarwal** [comment]
```
$ opam list -i
# Installed packages for ocaml.org:
base-bigarray        base  Bigarray library distributed with the OCaml compiler
base-bytes           base  Bytes library distributed with the OCaml compiler
base-threads         base  Threads library distributed with the OCaml compiler
base-unix            base  Unix library distributed with the OCaml compiler
camlp4           4.02.1+1  Camlp4 is a system for writing extensible parsers for programming languages
cmdliner            0.9.6  Declarative definition of command line interfaces for OCaml
cow                 1.1.0  XML, JSON, HTML, CSS, and Markdown syntax and libraries
cudf                  0.7  CUDF library (part of the Mancoosi tools)
dose           3.2.2+opam  Dose library (part of Mancoosi tools)
dyntype             0.9.0  syntax extension which makes OCaml types and values easier to manipulate programmatically
extlib-compat       1.6.1  A complete yet small extension for OCaml standard library (full, compatibility)
ezjsonm             0.4.0  An easy interface on top of the Jsonm library
hex                 0.1.0  Hexadecimal converter
js_of_ocaml           2.5  Compiler from OCaml bytecode to Javascript
jsonm               0.9.1  Non-blocking streaming JSON codec for OCaml
lwt                 2.4.7  A cooperative threads library for OCaml
menhir           20141215  LR(1) parser generator
mpp                 0.1.5  A preprocessor meant to blend languages.
ocamlfind           1.5.5  A library manager for OCaml
ocamlgraph          1.8.5  A generic graph library for OCaml
ocamlnet            3.7.7  Internet protocols (http, cgi, email etc.) and helper data structures (mail messages, character sets, etc.)
ocamlrss            2.2.2  Library providing functions to parse and print RSS 2.0 files
omd                 1.2.5  A Markdown frontend in pure OCaml.
opam-lib            1.2.0  The OCaml PAckage Manager (OPAM)
opam2web            1.4.0  A tool to generate a website from an OPAM repository
opamfu              0.1.2  Functions over OPAM Universes
ounit               2.0.0  Unit testing framework loosely based on HUnit. It is similar to JUnit, and other XUnit testing frameworks
ppx_tools          0.99.2  Tools for authors of ppx rewriters and other syntactic tools
re                  1.2.2  RE is a regular expression library for OCaml
sexplib         112.06.00  Library for serializing OCaml values to and from S-expressions
ssl                 0.4.7  Bindings for OpenSSL
stringext           1.2.0  Extra string functions for OCaml
type_conv       112.01.00  Library for building type-driven syntax extensions
ulex                  1.1  lexer generator for Unicode and OCaml
uri                 1.7.2  RFC3986 URI/URL parsing library
uutf                0.9.3  Non-blocking streaming Unicode codec for OCaml
xmlm                1.2.0  Streaming XML codec for OCaml
```

**Comment by Chris00** [comment]
@agarwal did you remove your pinned packages (including `opam-libs`)?

**Comment by agarwal** [comment]
Yes.

```
$ opam pin
ocamlorg.2.1.0            path     /Users/ashish/code/ocaml.org
```

**Comment by dsheets** [comment]
Perhaps camlp4 has had a breaking change? iirc, I used 4.01.0 to test this stuff and maybe 4.02.x or 4.02.1 or 4.02.1+1 are different.

Someone should also look at the resulting site build and ensure that the opam2web widget on the front page works correctly, still.

**Comment by Chris00** [comment]
It indeeds compile with OCaml 4.01.0 and fails with 4.02.1.  I switched to 4.01.0 on the server.

**Comment by agarwal** [comment]
Okay, so I'll add to the opam file:

```
available: [ ocaml-version = "4.01.0" ]
```

**Comment by agarwal** [comment]
I tested with 4.01.0 and this solves the problem for me too. Yay! I'm finally building the site again after a long time. @dsheets and @Chris00 Thanks for all your help.
